Improve Status output for Shadow Link

michael-redpanda opened the pull request:
<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

Improves status information retrieved in the `ShadowLink` resource type.

* Adds in status of tasks running within the link
* Sets the key fingerprint in the status
* Sets the "password last updated" field

Fixes: CORE-14281

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [X] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.2.x
- [ ] v25.1.x
- [ ] v24.3.x

## Release Notes

<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->

* None

michael-redpanda committed:
cl: Add method to get task status report by link

Signed-off-by: Michael Boquard <michael@redpanda.com>

michael-redpanda committed:
cl/model: Update shadow link status report to include tasks

Signed-off-by: Michael Boquard <michael@redpanda.com>

michael-redpanda committed:
cl: Add support for new fields in task report

Signed-off-by: Michael Boquard <michael@redpanda.com>

michael-redpanda committed:
cl: Update shadow link report to include task status

Signed-off-by: Michael Boquard <michael@redpanda.com>

michael-redpanda committed:
proto: Add shard to the task status report

Signed-off-by: Michael Boquard <michael@redpanda.com>

michael-redpanda committed:
admin/sl: Hook in task status report

Signed-off-by: Michael Boquard <michael@redpanda.com>

michael-redpanda committed:
cl: Improve state transitions for security migrator

Transition to link unavailable if there is a broker error when
attempting to get ACLs from the source cluster.

Signed-off-by: Michael Boquard <michael@redpanda.com>

michael-redpanda committed:
dt: Add tests to check state status

Signed-off-by: Michael Boquard <michael@redpanda.com>

michael-redpanda committed:
admin/sl: Provide key fingerprint

Also do not return the key value

Signed-off-by: Michael Boquard <michael@redpanda.com>

michael-redpanda committed:
dt: Update TLS tests to verify returned status

Signed-off-by: Michael Boquard <michael@redpanda.com>

michael-redpanda committed:
cl/model: Saving time since last password update

Signed-off-by: Michael Boquard <michael@redpanda.com>

michael-redpanda committed:
admin/sl: Wire in last password update timestamp

The last password set timestamp will be updated whenever a shadow link
is created with a set of SCRAM credentials or whenever the user updates
the SCRAM credentials used by the shadow link.

Signed-off-by: Michael Boquard <michael@redpanda.com>

michael-redpanda committed:
dt: Test SCRAM user status

Signed-off-by: Michael Boquard <michael@redpanda.com>

michael-redpanda commented:
Force push:

* Fixed git history
* Removed debug log lines

michael-redpanda committed:
cl/model: Remove unused precision from time format

Signed-off-by: Michael Boquard <michael@redpanda.com>

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/74878-->
<!--e5de9d5d-6a22-4c69-97ad-b4647996f31e-->
<!--05753ad8-a247-43a1-b789-11f3772e9874-->
<!--8c1e16c5-a3b8-41e3-88d1-35817926ef3c-->
<!--cbfabf30-c22f-4af1-916a-a5469eddb83a-->
<!--title-->
## Retry command for [Build#74878](https://buildkite.com/redpanda/redpanda/builds/74878)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/shadow_linking_rnot_test.py::ShadowLinkingRandomOpsTest.test_node_operations@{"failures":false}
tests/rptest/tests/shadow_linking_rnot_test.py::ShadowLinkingRandomOpsTest.test_node_operations@{"failures":true}
```
<!--/body-->

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--5b31c05a-e596-4d9c-8b38-203a8b01658f-->
<!--TEST RESULTS-->
<!--db888ce8-8274-4ebc-bb2c-078d1022b288-->
<!--TEST RESULTS-->
<!--20701ea6-f266-4427-829c-abd1732a9616-->
<!--TEST RESULTS-->
<!--dc87d382-6b1a-4f65-b4cf-bbec4b307652-->
<!--TEST RESULTS-->
<!--5a63324e-4df8-4dcf-aa4e-5da600c939d4-->
<!--TEST RESULTS-->
<!--b97f724e-ba66-48b3-8222-f164b0339cc5-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/74878">build#74878</a></summary>


| test_class                 | test_method                                             | test_arguments                               | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  | test_history                                                                                                                                                                                   |
|:---------------------------|:--------------------------------------------------------|:---------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| MountUnmountIcebergTest    | test_simple_remount                                     | {"cloud_storage_type": 1}                    | integration | https://buildkite.com/redpanda/redpanda/builds/74878#019a13db-e9f6-447f-a668-846d0c8c0cd4 | FLAKY         | 15/21    | upstream reliability is '100.0'. current run reliability is '71.42857142857143'. drift is 28.57143 and the allowed drift is set to 50. The test should PASS             | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=MountUnmountIcebergTest&test_method=test_simple_remount                                      |
| NodesDecommissioningTest   | test_decommissioning_finishes_after_manual_cancellation | {"cloud_topic": false, "delete_topic": true} | integration | https://buildkite.com/redpanda/redpanda/builds/74878#019a13db-e9fb-4d94-b847-a922ee391649 | FLAKY         | 20/21    | upstream reliability is '98.21428571428571'. current run reliability is '95.23809523809523'. drift is 2.97619 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=NodesDecommissioningTest&test_method=test_decommissioning_finishes_after_manual_cancellation |
| PartitionMovementTest      | test_empty                                              | {"num_to_upgrade": 0}                        | integration | https://buildkite.com/redpanda/redpanda/builds/74878#019a13db-e9f6-447f-a668-846d0c8c0cd4 | FLAKY         | 18/21    | upstream reliability is '95.86330935251799'. current run reliability is '85.71428571428571'. drift is 10.14902 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=PartitionMovementTest&test_method=test_empty                                                 |
| ShadowLinkingRandomOpsTest | test_node_operations                                    | {"failures": false}                          | integration | https://buildkite.com/redpanda/redpanda/builds/74878#019a13da-ba18-424d-be80-86f75273efe8 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingRandomOpsTest&test_method=test_node_operations                                  |
| ShadowLinkingRandomOpsTest | test_node_operations                                    | {"failures": false}                          | integration | https://buildkite.com/redpanda/redpanda/builds/74878#019a13db-e9fc-410b-8de5-d6387ee202df | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingRandomOpsTest&test_method=test_node_operations                                  |
| ShadowLinkingRandomOpsTest | test_node_operations                                    | {"failures": true}                           | integration | https://buildkite.com/redpanda/redpanda/builds/74878#019a13da-ba19-4499-a4e1-0ba3115564a6 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingRandomOpsTest&test_method=test_node_operations                                  |
| ShadowLinkingRandomOpsTest | test_node_operations                                    | {"failures": true}                           | integration | https://buildkite.com/redpanda/redpanda/builds/74878#019a13db-e9fd-4cb2-ad3a-1b2a8ba4a94c | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingRandomOpsTest&test_method=test_node_operations                                  |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/74904">build#74904</a></summary>


| test_class                                   | test_method                                  | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  | test_history                                                                                                                                                             |
|:---------------------------------------------|:---------------------------------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ShadowLinkBasicTests                         | test_rapid_shadow_link_toggling              | null             | integration | https://buildkite.com/redpanda/redpanda/builds/74904#019a162d-261c-4da8-99af-27035c972c64 | FLAKY         | 16/21    | upstream reliability is '91.98542805100182'. current run reliability is '76.19047619047619'. drift is 15.79495 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkBasicTests&test_method=test_rapid_shadow_link_toggling       |
| DataMigrationsApiTest                        | test_creating_and_listing_migrations         | null             | integration | https://buildkite.com/redpanda/redpanda/builds/74904#019a162d-2615-4982-920f-b95d221c1898 | FAIL          | 0/1      |                                                                                                                                                                         | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataMigrationsApiTest&test_method=test_creating_and_listing_migrations |
| src/v/storage/tests/storage_e2e_fixture_test | src/v/storage/tests/storage_e2e_fixture_test |                  | unit        | https://buildkite.com/redpanda/redpanda/builds/74904#019a15fd-f04e-4be9-9ab4-fa8d6e6b4563 | FAIL          | 0/1      |                                                                                                                                                                         |                                                                                                                                                                          |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75021">build#75021</a></summary>


| test_class                                   | test_method                                  | test_arguments                                                                                                 | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  | test_history                                                                                                                                                              |
|:---------------------------------------------|:---------------------------------------------|:---------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:--------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ShadowLinkBasicTests                         | test_rapid_shadow_link_toggling              | null                                                                                                           | integration | https://buildkite.com/redpanda/redpanda/builds/75021#019a25ab-6132-443e-97da-006a56093caf | FLAKY         | 12/21    | upstream reliability is '90.35621198957429'. current run reliability is '57.14285714285714'. drift is 33.21335 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkBasicTests&test_method=test_rapid_shadow_link_toggling        |
| ShadowLinkBasicTests                         | test_task_states_change                      | null                                                                                                           | integration | https://buildkite.com/redpanda/redpanda/builds/75021#019a25d5-34d6-4c20-b027-910156017e21 | FLAKY         | 14/21    | The test was found to be new, and no failures are allowed                                                                                                               | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkBasicTests&test_method=test_task_states_change                |
| ShadowLinkBasicTests                         | test_topic_properties_replicated             | {"source_cluster_spec": {"cluster_type": "kafka", "kafka_quorum": "COMBINED_KRAFT", "kafka_version": "3.8.0"}} | integration | https://buildkite.com/redpanda/redpanda/builds/75021#019a25ab-612e-4432-b48a-e4a52e1696f0 | FLAKY         | 20/21    | upstream reliability is '94.68085106382979'. current run reliability is '95.23809523809523'. drift is -0.55724 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkBasicTests&test_method=test_topic_properties_replicated       |
| DataMigrationsApiTest                        | test_creating_and_listing_migrations         | null                                                                                                           | integration | https://buildkite.com/redpanda/redpanda/builds/75021#019a25ab-612d-48af-94bd-68732c52be18 | FLAKY         | 17/21    | upstream reliability is '99.70631424375918'. current run reliability is '80.95238095238095'. drift is 18.75393 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataMigrationsApiTest&test_method=test_creating_and_listing_migrations  |
| NodesDecommissioningTest                     | test_recommissioning_node_finishes           | {"cloud_topic": false}                                                                                         | integration | https://buildkite.com/redpanda/redpanda/builds/75021#019a25d5-34d2-4c19-b209-d64be99cb00f | FLAKY         | 20/21    | upstream reliability is '98.63325740318906'. current run reliability is '95.23809523809523'. drift is 3.39516 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=NodesDecommissioningTest&test_method=test_recommissioning_node_finishes |
| TopicRecoveryTest                            | test_many_partitions                         | {"check_mode": "check_manifest_existence", "cloud_storage_type": 2}                                            | integration | https://buildkite.com/redpanda/redpanda/builds/75021#019a25ab-6131-460e-b39e-212eae8c4ebf | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS               | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=TopicRecoveryTest&test_method=test_many_partitions                      |
| src/v/storage/tests/storage_e2e_fixture_test | src/v/storage/tests/storage_e2e_fixture_test |                                                                                                                | unit        | https://buildkite.com/redpanda/redpanda/builds/75021#019a257c-aff0-42fe-b0a7-7b6bc1640ee0 | FAIL          | 0/1      |                                                                                                                                                                         |                                                                                                                                                                           |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75059">build#75059</a></summary>


| test_class                 | test_method                          | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  | test_history                                                                                                                                                             |
|:---------------------------|:-------------------------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| DataMigrationsApiTest      | test_creating_and_listing_migrations | null             | integration | https://buildkite.com/redpanda/redpanda/builds/75059#019a2733-3623-4e81-ade6-205eabc12ff8 | FLAKY         | 18/21    | upstream reliability is '99.56011730205279'. current run reliability is '85.71428571428571'. drift is 13.84583 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataMigrationsApiTest&test_method=test_creating_and_listing_migrations |
| LogCompactionTxRemovalTest | test_tx_control_batch_removal        | null             | integration | https://buildkite.com/redpanda/redpanda/builds/75059#019a272d-ce07-4f7f-b1a2-62fe1a5ce1c5 | FLAKY         | 16/21    | upstream reliability is '100.0'. current run reliability is '76.19047619047619'. drift is 23.80952 and the allowed drift is set to 50. The test should PASS             | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalTest&test_method=test_tx_control_batch_removal   |
| TLSVersionTestRSA          | test_change_version                  | {"version": 1}   | integration | https://buildkite.com/redpanda/redpanda/builds/75059#019a272d-ce02-492d-9331-ae3845f680b5 | FAIL          | 0/1      |                                                                                                                                                                         | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=TLSVersionTestRSA&test_method=test_change_version                      |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75102">build#75102</a></summary>


| test_class                        | test_method                                | test_arguments                    | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 | test_history                                                                                                                                                                               |
|:----------------------------------|:-------------------------------------------|:----------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| LogCompactionTxRemovalUpgradeTest | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "All commits"} | integration | https://buildkite.com/redpanda/redpanda/builds/75102#019a2ae1-ae3d-4969-bad0-21e91d968160 | FLAKY         | 19/21    | upstream reliability is '96.82539682539682'. current run reliability is '90.47619047619048'. drift is 6.34921 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| TLSVersionTestRSA                 | test_change_version                        | {"version": 1}                    | integration | https://buildkite.com/redpanda/redpanda/builds/75102#019a2adb-fbc6-4b5f-982f-a57db1f458e9 | FAIL          | 0/1      |                                                                                                                                                                        | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=TLSVersionTestRSA&test_method=test_change_version                                        |
| TopicDeleteStressTest             | stress_test                                | null                              | integration | https://buildkite.com/redpanda/redpanda/builds/75102#019a2adb-fbc9-4a7c-bcdd-8c9e10a1d9c2 | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS              | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=TopicDeleteStressTest&test_method=stress_test                                            |
| ProducersAdminAPITest             | test_producers_state_api_during_load       | null                              | integration | https://buildkite.com/redpanda/redpanda/builds/75102#019a2adb-fbc9-4a7c-bcdd-8c9e10a1d9c2 | FLAKY         | 20/21    | upstream reliability is '99.76470588235294'. current run reliability is '95.23809523809523'. drift is 4.52661 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ProducersAdminAPITest&test_method=test_producers_state_api_during_load                   |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75433">build#75433</a></summary>


| test_class                    | test_method                       | test_arguments                                                                                                                                                 | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  | test_history                                                                                                                                                                  |
|:------------------------------|:----------------------------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ShadowLinkingReplicationTests | test_replication_timestamps_match | {"source_cluster_spec": {"cluster_type": "redpanda"}, "timestamp_type": "CreateTime"}                                                                          | integration | https://buildkite.com/redpanda/redpanda/builds/75433#019a3b69-48ba-4b9c-9491-4005b153859c | FLAKY         | 20/21    | upstream reliability is '95.28130671506352'. current run reliability is '95.23809523809523'. drift is 0.04321 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingReplicationTests&test_method=test_replication_timestamps_match |
| ShadowLinkingReplicationTests | test_replication_timestamps_match | {"source_cluster_spec": {"cluster_type": "redpanda"}, "timestamp_type": "CreateTime"}                                                                          | integration | https://buildkite.com/redpanda/redpanda/builds/75433#019a3b6d-8e32-4b5e-b2ae-318cb9160e05 | FLAKY         | 20/21    | upstream reliability is '95.1178451178451'. current run reliability is '95.23809523809523'. drift is -0.12025 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingReplicationTests&test_method=test_replication_timestamps_match |
| LogCompactionTxRemovalTest    | test_tx_control_batch_removal     | null                                                                                                                                                           | integration | https://buildkite.com/redpanda/redpanda/builds/75433#019a3b69-48c0-4095-8f21-58c794f58dd7 | FLAKY         | 7/21     | upstream reliability is '89.0909090909091'. current run reliability is '33.33333333333333'. drift is 55.75758 and the allowed drift is set to 50. The test should FAIL  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalTest&test_method=test_tx_control_batch_removal        |
| PartitionMovementTest         | test_empty                        | {"num_to_upgrade": 0}                                                                                                                                          | integration | https://buildkite.com/redpanda/redpanda/builds/75433#019a3b6d-8e3a-456c-8c58-c92c02f2f398 | FLAKY         | 19/21    | upstream reliability is '87.27695888285493'. current run reliability is '90.47619047619048'. drift is -3.19923 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=PartitionMovementTest&test_method=test_empty                                |
| TieredStorageTest             | test_tiered_storage               | {"cloud_storage_type_and_url_style": [2, "virtual_host"], "test_case": {"name": "(TS_Read == True, TS_Timequery == True, SpilloverManifestUploaded == True)"}} | integration | https://buildkite.com/redpanda/redpanda/builds/75433#019a3b69-48bb-4195-9651-9d9a0dc9d477 | FLAKY         | 20/21    | upstream reliability is '99.51100244498777'. current run reliability is '95.23809523809523'. drift is 4.27291 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=TieredStorageTest&test_method=test_tiered_storage                           |
</details>
<!--/body-->

IoannisRP commented on src/v/cluster_link/service.cc:
```diff
@@ -1324,6 +1331,14 @@ service::shard_local_shadow_link_report(model::id_t id) {
     if (auto err = check_manager_state(); err != errc::success) {
         return rpc::shadow_link_status_report_response{.err_code = err};
     }
+    auto on_controller_leader = [this]() -> bool {
+        if (ss::this_shard_id() != cluster::controller_stm_shard) {
+            return false;
+        }
+        return _self
+               == _partition_leaders_table->local().get_leader(
+                 ::model::controller_ntp);
+    }();
```
nit: could this asks the `_manager` about controller leadership? We already track it there.

michael-redpanda commented:
/ci-repeat 1

rockwotj commented on proto/redpanda/core/admin/v2/shadow_link.proto:
```diff
@@ -535,6 +535,8 @@ message ShadowLinkTaskStatus {
     // The broker the task is running on
     int32 broker_id = 4;
+    // The shard the task is running on
+    int32 shard = 5;
```
Probably fine, but we have talked about in the past not exposing shard level information in cloud, but I get that this is useful to know.

michael-redpanda replied:
yeah I agree in principal but in this situation since we have tasks running per shard I think it would be helpful to know if certain tasks are failing

rockwotj reviewed (dismissed):
Only reviewed the protos, lgtm

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/74904-->
<!--115b9643-6370-4c44-9a13-0919be45f1f7-->
<!--title-->
## Retry command for [Build#74904](https://buildkite.com/redpanda/redpanda/builds/74904)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/data_migrations_api_test.py::DataMigrationsApiTest.test_creating_and_listing_migrations
```
<!--/body-->

michael-redpanda commented:
Force push:

* Rebased off of dev

rockwotj commented on src/v/cluster_link/model/types.h:
```diff
@@ -199,10 +200,15 @@ struct scram_credentials
     /// SASL-SCRAM mechanism to use
     ss::sstring mechanism;
+    /// This records when the password was last updated.  Can not be a time
+    /// point since we don't serialize it so it'll be duration since epoch
```
We don't serialize it?

michael-redpanda replied:
time points?  no: https://github.com/redpanda-data/redpanda/blob/adceb688188eabe583ea1c41f035a25a26cd7953/src/v/serde/rw/chrono.h#L116-L133

bharathv replied:
nit:

> This records when the password was last updated.

I think the comment wording is a bit misleading.. It implies it is a timepoint when actually its a duration.. It should probably say "time since the last password update or sth like that".

michael-redpanda replied:
It's the time since epoch that the password was set - not the duration since the last change in password

rockwotj reviewed (dismissed):
protos lgtm

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/75021-->
<!--7f16adde-a284-4e87-b883-82240789aa76-->
<!--title-->
## Retry command for [Build#75021](https://buildkite.com/redpanda/redpanda/builds/75021)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/cluster_linking_e2e_test.py::ShadowLinkBasicTests.test_task_states_change
```
<!--/body-->

michael-redpanda committed:
cl: Adjust log level

`error` level too high for when a source cluster doesn't support a
specific API.

Signed-off-by: Michael Boquard <michael@redpanda.com>

michael-redpanda commented:
Force push:

* Reorded some commits
* Adjust log level from error -> warn

michael-redpanda commented:
Force push:

* Fixed code lint

joe-redpanda commented on src/v/redpanda/admin/services/shadow_link/converter.cc:
```diff
@@ -517,6 +521,10 @@ authentication_configuration create_authentication_configuration(
           scram_proto.set_password_set(true);
+          scram_proto.set_password_set_at(
+            absl::FromUnixNanos(
+              absl::ToInt64Nanoseconds(
```
q: 
time point -> absl::Time -> epoch nanos -> absl::Time again?

michael-redpanda replied:
It's not a time point it's a duration.  But yeah going from absl::Duration to absl::Time can be done better:
`absl::UnixEpoch() + absl::FromChrono(scram.password_last_updated)`

joe-redpanda reviewed (dismissed):
question otherwise lgtm

dotnwat commented on src/v/redpanda/admin/services/shadow_link/converter.cc:
```diff
@@ -1062,6 +1113,60 @@ void merge_output_only_fields(
+                      return;
+                  }
+                  c.password_last_updated
+                    = ss::lowres_clock::now().time_since_epoch();
```
AFAICT `password_last_updated` is intended to be wall-clock time (i.e. "when was the password last updated" should have some kind of unambiguous meaning). however, the `time_since_epoch` here, IIRC, will be in reference not to the Unix epoch but rather an undefined epoch (usually time-since-boot). So it seems like maybe you want to use `lowres_system_clock` here (and several other call sites)?

michael-redpanda replied:
oh good call thanks

michael-redpanda commented:
Force push:

* Addressed PR comments

bharathv commented on src/v/cluster_link/model/types.cc:
```diff
@@ -351,20 +351,25 @@ shadow_link_status_topic_response::format_to(fmt::iterator it) const {
+            return fmt::format("topic_name: {}, status: {}", t.first, t.second);
+        }),
+        ", "),
+      fmt::join(
+        task_status_reports | std::views::transform([](auto& t) {
+            return fmt::format(
+              "task_name: {}, statuses: {}",
+              t.first,
+              fmt::join(t.second.begin(), t.second.end(), ", "));
+        }),
+        ", "));
```
My usual rant about overallocs :P

michael-redpanda replied:
Yes, however in this situation we have a fairly limited number of tasks that exist

bharathv replied:
I think cumulative string is still a problem (including topic reports). (obviously not expecting to be fixed in this PR).

bharathv commented on src/v/cluster_link/model/types.cc:
```diff
@@ -405,11 +405,17 @@ auto fmt::formatter<cluster_link::model::task_state>::format(
+    auto time_point = std::chrono::system_clock::time_point{
+      std::chrono::duration_cast<std::chrono::system_clock::duration>(
+        c.password_last_updated)};
+    auto time = std::format("{:%FT%H:%M:%S}", time_point);
```
Is this correct? I think this timepoint{duration} c'tor means  system_clock_epoch (= ~1970) + duration.

https://en.cppreference.com/w/cpp/chrono/time_point/time_point.html

edit: scratch my comment, maybe this is correct, it is always saved as system_clock::time_since_epoch() (for some reason I was under the impression that it was time_since_last_update", so computing from epoch seemed wrong).

michael-redpanda replied:
Sorry for the confusion - yes the last time it was updated as a timepoint but since we don't serialize time points it's set as a duration since last epoch.  Reconcilers can read this back to know the last time the password was updated (as a time stamp)

dotnwat replied:
> but since we don't serialize time points it's set as a duration since last epoch

isn't a "duration since some epoch" the same thing as a time point?

michael-redpanda replied:
effectively?  I'm stating that we don't serialize `std::chrono::time_point` in serde https://github.com/redpanda-data/redpanda/blob/adceb688188eabe583ea1c41f035a25a26cd7953/src/v/serde/rw/chrono.h#L116-L133

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/75059-->
<!--f87de69a-70c7-4432-ac42-df00b2c79221-->
<!--title-->
## Retry command for [Build#75059](https://buildkite.com/redpanda/redpanda/builds/75059)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/tls_version_test.py::TLSVersionTestRSA.test_change_version@{"version":1}
```
<!--/body-->

dotnwat commented on src/v/cluster_link/model/types.h:
```diff
@@ -199,10 +200,15 @@ struct scram_credentials
     ss::sstring mechanism;
+    /// This records when the password was last updated.  Can not be a time
+    /// point since we don't serialize it so it'll be duration since epoch
+    ss::lowres_clock::duration password_last_updated;
```
> /// This records when the password was last updated.  Can not be a time
> /// point since we don't serialize it so it'll be duration since epoch

I think a time_point is nothing more than a duration since an epoch, so all this does is circumvent the technical limitation of the code base not having a serde wrapper for the chrono::time_point type. The reason we try to discourage serializing time points is that we want to discourage comparing time across possibly unsynchronized clocks.

But in reality we do serialize time points, the most common case is `model::timestamp`. I think in this particular case, where I'm guessing but correct me if I'm wrong, that this is more of a UX convenience thing, it seems fine to serialize a time point.

michael-redpanda replied:
> But in reality we do serialize time points, the most common case is model::timestamp

Gotcha, I did not realize that.  Ran into issues when just using `std::chrono::time_point`.  I can use `model::timestamp` instead.

> that this is more of a UX convenience thing

Correct.  In talking with the k8s team, they were looking for some feedback regarding if the password needed updating during a reconciliation loop.  We came to an agreement on using a wall-clock timestamp.

michael-redpanda commented:
Force push:

* Used `model::timestamp`

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/75102-->
<!--d0b485bb-5e3a-47d4-95a8-f498582166df-->
<!--title-->
## Retry command for [Build#75102](https://buildkite.com/redpanda/redpanda/builds/75102)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/tls_version_test.py::TLSVersionTestRSA.test_change_version@{"version":1}
```
<!--/body-->

kbatuigas commented:
Docs is good with the protos, t.y.

michael-redpanda commented:
Force push:

* Rebased off of dev

michael-redpanda commented:
Force push:

* Fixed issue with rebase

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/75433-->
<!--04041c15-689f-44c3-8c37-25c8b9120e93-->
<!--title-->
## Retry command for [Build#75433](https://buildkite.com/redpanda/redpanda/builds/75433)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/log_compaction_test.py::LogCompactionTxRemovalTest.test_tx_control_batch_removal
```
<!--/body-->

michael-redpanda commented:
CI Failures:

* https://redpandadata.atlassian.net/browse/CORE-14444
